python/cuml/naive_bayes/naive_bayes.py

```diff
@@ -0,0 +1,533 @@
+            self.fit_called_ = True
+            if _classes is not None:
+                check_labels(Y, _classes)
+                self.classes_ = _classes
+            else:
+                self.classes_ = label_classes
+
+            self.n_classes_ = self.classes_.shape[0]
+            self.n_features_ = X.shape[1]
+            self._init_counters(self.n_classes_, self.n_features_,
+                                X.dtype)
+        else:
+            check_labels(Y, self.classes_)
+
+        self._count(X, Y)
+
+        self._update_feature_log_prob(self.alpha)
+        self._update_class_log_prior(class_prior=self.class_prior)
+
+        return self
+
+    def update_log_probs(self):
+        """
+        Updates the log
+        :return:
+        """
+        self._update_feature_log_prob(self.alpha)
+        self._update_class_log_prior(class_prior=self.class_prior)
+
+    def partial_fit(self, X, y, classes=None, sample_weight=None):
+        """
+        Incremental fit on a batch of samples.
+
+        This method is expected to be called several times consecutively on
+        different chunks of a dataset so as to implement out-of-core or online
+        learning.
+
+        This is especially useful when the whole dataset is too big to fit in
+        memory at once.
+
+        This method has some performance overhead hence it is better to call
+        partial_fit on chunks of data that are as large as possible (as long
+        as fitting in the memory budget) to hide the overhead.
+
+        Parameters
+        ----------
+
+        X : {array-like, cupy sparse matrix} of shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features
+
+        y : array-like of shape (n_samples) Target values.
+        classes : array-like of shape (n_classes)
+                  List of all the classes that can possibly appear in the y
+                  vector. Must be provided at the first call to partial_fit,
+                  can be omitted in subsequent calls.
+
+        sample_weight : array-like of shape (n_samples)
+                        Weights applied to individual samples (1. for
+                        unweighted). Currently sample weight is ignored
+
+        Returns
+        -------
+
+        self : object
+        """
+        return self._partial_fit(X, y, sample_weight=sample_weight,
+                                 _classes=classes)
+
+    @cp.prof.TimeRangeDecorator(message="predict()", color_id=1)
+    def predict(self, X):
+
+        """
+        Perform classification on an array of test vectors X.
+
+        Parameters
+        ----------
+
+        X : array-like of shape (n_samples, n_features)
+
+        Returns
+        -------
+
+        C : cupy.ndarray of shape (n_samples)
+
+        """
+
+        jll = self._joint_log_likelihood(X)
+        indices = cp.argmax(jll, axis=1).astype(self.classes_.dtype)
```

Comment:
I might be misunderstanding the use of `rmm_cupy_ary` but I thought it only needed to be used for new buffer allocations in arrays that weren't just wrapping another array. I've been assuming CuPy in inheriting the allocator from wrapped classes, otherwise it would be strange if it's running kernels on memory that's allocated in two different pools. We might need to check what it's doing under the hood.